dipy/align/transforms.pyx

```diff
@@ -0,0 +1,955 @@
+        if self.number_of_parameters < 0:
+            raise ValueError("Invalid transform.")
+        theta = np.zeros(self.number_of_parameters)
+        self._get_identity_parameters(theta)
+        return theta
+
+    def param_to_matrix(self, double[:] theta):
+        r""" Matrix representation of this transform with the given parameters
+
+        Parameters
+        ----------
+        theta : array, shape (n,)
+            the parameter values of the transform
+
+        Returns
+        -------
+        T : array, shape (dim + 1, dim + 1)
+            the matrix representation of this transform with parameters theta
+        """
+        n = len(theta)
+        if n != self.number_of_parameters:
+            raise ValueError("Invalid number of parameters: %d"%(n,))
+        T = np.eye(self.dim + 1)
+        self._param_to_matrix(theta, T)
+        return T
+
+    def get_number_of_parameters(self):
+        return self.number_of_parameters
+
+    def get_dim(self):
+        return self.dim
+
+
+cdef class TranslationTransform2D(Transform):
+    def __init__(self):
+        r""" Translation transform in 2D
+        """
+        self.dim = 2
+        self.number_of_parameters = 2
+
+    cdef int _jacobian(self, double[:] theta, double[:] x, double[:,:] J) nogil:
+        r""" Jacobian matrix of the 2D translation transform
+        The transformation is given by:
+
+        T(x) = (T1(x), T2(x)) = (x0 + t0, x1 + t1)
+
+        The derivative w.r.t. t1 and t2 is given by
+
+        T'(x) = [[1, 0], # derivatives of [T1, T2] w.r.t. t0
+                 [0, 1]] # derivatives of [T1, T2] w.r.t. t1
+
+        Parameters
+        ----------
+        theta : array, shape(2,)
+            the parameters of the 2D translation transform (the Jacobian does not
+            depend on the parameters, but we receive the buffer so all Jacobian
+            functions receive the same parameters)
+        x : array, shape(2,)
+            the point at which to compute the Jacobian (the Jacobian does not
+            depend on x, but we receive the buffer so all Jacobian functions
+            receive the same parameters)
+        J : array, shape(2, 2)
+            the buffer in which to write the Jacobian
+
+        Returns
+        -------
+        is_constant : int
+            always returns 1, indicating that the Jacobian is constant
+            (independent of x)
+        """
+        J[0,0], J[0, 1] = 1.0, 0.0
+        J[1,0], J[1, 1] = 0.0, 1.0
+        # This Jacobian does not depend on x (it's constant): return 1
+        return 1
+
+    cdef void _get_identity_parameters(self, double[:] theta) nogil:
+        r""" Parameter values corresponding to the identity
+        Sets in theta the parameter values corresponding to the identity
+        transform
+
+        Parameters
+        ----------
+        theta : array, shape(2,)
+            buffer to write the parameters of the 2D translation transform
+        """
+        theta[:2] = 0
```

Comment:
Is this really applied properly in Cython?
So, this will do theta[0] = 0 and theta[1] = 1? No for loop needed?